File: Solutions/347- Top K Frequent Elements/Solution.java
import java.util.Hashtable;
import java.util.Map.Entry;
import java.util.PriorityQueue;

class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Hashtable<Integer, Integer> map = new Hashtable<>();
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        int[] answer = new int[k];

        for (int num : nums) {
            if (map.get(num) == null)
                map.put(num, 1);
            else
                map.put(num, map.get(num) + 1);
        }

        for (var value : map.values()) {
            queue.add(value);
            if (queue.size() > k)
                queue.remove();
        }

        int index = 0;
        while (!queue.isEmpty()) {
            int item = queue.remove();
            for (Entry<Integer, Integer> entry : map.entrySet()) {
                if (entry.getValue() == item) {
                    answer[index] = entry.getKey();
                    map.remove(entry.getKey());
                    index++;
                    break;
                }
            }
        }

        return answer;
    }
}

File: Solutions/347- Top K Frequent Elements/Solution.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}
        answer = list()

        for num in nums:
            if map.get(num) is None:
                map[num] = 1
            else:
                map[num] += 1

        ordered_map = sorted(map.items(), key=lambda item: item[1], reverse=True)
        top_k_items =  ordered_map[:k]

        for key, value in top_k_items:
            answer.append(key)


        return answer

